addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4222,7 +4222,8 @@ registry.BackgroundShape = SnippetOptionWidget.extend({
                 shapeToSelect = possibleShapes.find((shape, i) => {
                     return possibleShapes[i - 1] === previousShape;
                 });
-            } else {
+            }
+            if (!shapeToSelect) {
```

Comment:
It's worth a small comment to explain the two cases this new `if` covers IMO (the previous else and the find which did not return anything :+1:)